src/main/webapp/ratingScript.js

```diff
@@ -110,6 +112,34 @@ async function getLatestRating() {
   }
 }
 
+function validSubmission() {
+  const allFormFields = [
+    document.getElementById("term-input"),
+    document.getElementById("prof-input"),
+    document.getElementById("rating-term"),
+    document.getElementById("rating-prof"),
+    document.getElementById("hours"),
+    document.getElementById("difficulty"),
+    document.getElementById("grade"),
+  ];
+
+  let i;
+  let successfulSubmissionMessage = document.getElementById(
+    "sucessful-submission-message"
+  );
+  for (i = 0; i < allFormFields.length; i++) {
+    // If any field is not field out properly.
+    if (!allFormFields[i].checkValidity()) {
+      successfulSubmissionMessage.innerHTML =
```

Comment:
It's a little odd that a function returning a boolean indicating whether or not form fields are valid also has a side effect writing a message to the dom. Might want to pull out the side effect.

Reply:
Makes sense, thanks for the comment making all the changes now!